### PR TITLE
Fix/mascot - spectral batch size option - not yet ready

### PR DESCRIPTION
**Pull request by Leon-Bichmann**
# Description

This PR should introduce a spectra batch size option fo the Mascot Adapter for it to process only a given number of spectra at a time, in case the files are too large to be processed in one go.

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

### CODE REVIEW

**Comment by Leon-Bichmann** [comment]
Hey, so its not tested yet, I just opened it for now and will do further tests tomorrow.

**Comment by Leon-Bichmann** [comment]
ok seems like there is sth. wrong I get the following error when trying to build it locally:

```ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libOpenMS.dylib] Error 1
make[1]: *** [src/openms/CMakeFiles/OpenMS.dir/all] Error 2
make: *** [all] Error 2```

File: src/topp/MascotAdapter.cpp [hunk 1]
```diff
@@ -487,10 +487,35 @@ class TOPPMascotAdapter :
 
     if (!mascot_out)
     {
+    // determine number of chunks of size n
+    int size = (experiment.size() - 1) / n + 1;
+ 
+    // create array of vectors to store the chunks
+    std::vector<PeakMap> split_experiment[size];
```
Comment (Leon-Bichmann) [inline src/topp/MascotAdapter.cpp:496]: is std::vector<PeakMap> a valid type specification?

Comment (cbielow) [reply src/topp/MascotAdapter.cpp:496]: sure. If `vector` is `#include`d, either directly or indirectly :)

File: src/topp/MascotAdapter.cpp [hunk 2]
```diff
@@ -487,10 +489,35 @@ class TOPPMascotAdapter :
...
+
+    for (int k = 0; k < size; ++k)
+    {
+        // get range for next set of n elements
+        auto start_itr = std::next(experiment.cbegin(), k*n_chunks);
+        auto end_itr = std::next(experiment.cbegin(), k*n_chunks + n_chunks);
+
+        // allocate memory for the chunk
+        split_experiment[k].resize(n_chunks);
```
Comment (cbielow) [inline src/topp/MascotAdapter.cpp:505]: this is too much for the last bin, in most cases.

you can resize for all bins, based on `.resize(end_itr - start_itr);`

Comment (Leon-Bichmann) [reply src/topp/MascotAdapter.cpp:505]: alright, thx i'll take a look at it next week again

Comment (Leon-Bichmann) [reply src/topp/MascotAdapter.cpp:506]: ok so I thought the if clause would be fine to resize the last chunk. If I do it instead with the `end_itr - start_itr`. Can I use the same iterators twice?

**Comment by timosachsenberg** [comment]
wrong file ;)

**Pull request closed without merging**